typescript/src/services/deposits/deposits-service.ts

```diff
@@ -184,6 +368,347 @@ export class DepositsService {
+   *                             - "Sui" - Sui L2 deposits
+   *                             - "StarkNet" - StarkNet L2 deposits (note: capital 'N')
+   *                             Note: "Solana" is not currently supported for gasless deposits
+   * @returns GaslessDepositResult containing deposit object, receipt, and chain name
+   * @throws Throws an error if:
+   *         - Bitcoin recovery address is not P2PKH or P2WPKH
+   *         - Destination chain name is not in the supported list
+   *         - Destination chain contracts not initialized (for L2 deposits)
+   *         - NativeBTCDepositor address not available (for L1 deposits)
+   *         - Deposit owner cannot be resolved (for L2 deposits)
+   *         - No active wallet in Bridge contract
+   */
+  async initiateGaslessDeposit(
+    bitcoinRecoveryAddress: string,
+    destinationChainName: string
+  ): Promise<GaslessDepositResult> {
+    // Validate that the chain supports gasless deposits
+    if (!this.SUPPORTED_GASLESS_CHAINS.includes(destinationChainName as any)) {
+      throw new Error(
+        `Gasless deposits are not supported for chain: ${destinationChainName}. ` +
+          `Supported chains: ${this.SUPPORTED_GASLESS_CHAINS.join(", ")}`
+      )
+    }
+
+    // Validate Bitcoin recovery address early for consistent error behavior
+    const bitcoinNetwork = await this.bitcoinClient.getNetwork()
+    const recoveryOutputScript = BitcoinAddressConverter.addressToOutputScript(
+      bitcoinRecoveryAddress,
+      bitcoinNetwork
+    )
+    if (
+      !BitcoinScriptUtils.isP2PKHScript(recoveryOutputScript) &&
+      !BitcoinScriptUtils.isP2WPKHScript(recoveryOutputScript)
+    ) {
+      throw new Error("Bitcoin recovery address must be P2PKH or P2WPKH")
+    }
+
+    if (destinationChainName === "L1") {
+      return this.initiateL1GaslessDeposit(bitcoinRecoveryAddress)
+    } else {
+      return this.initiateL2GaslessDeposit(
+        bitcoinRecoveryAddress,
+        destinationChainName as DestinationChainName
+      )
+    }
+  }
+
+  /**
+   * Internal helper for L1 gasless deposits using NativeBTCDepositor.
+   * @param bitcoinRecoveryAddress - Bitcoin address for recovery if deposit fails (P2PKH or P2WPKH).
+   * @returns Promise resolving to GaslessDepositResult containing deposit, receipt, and "L1" chain name.
+   */
+  private async initiateL1GaslessDeposit(
+    bitcoinRecoveryAddress: string
+  ): Promise<GaslessDepositResult> {
+    let depositor = this.getNativeBTCDepositorAddress()
+    if (!depositor) {
+      depositor = await this.resolveNativeBTCDepositorFromNetwork()
+    }
+    if (!depositor) {
+      const network = await this.bitcoinClient.getNetwork()
+      throw new Error(
+        `NativeBTCDepositor address not available for Bitcoin network: ${network}`
+      )
+    }
+
+    const receipt = await this.generateDepositReceipt(
+      bitcoinRecoveryAddress,
+      depositor,
+      undefined
```

Comment:
Not sure how to address this, but maybe there is a way to avoid passing undefined here and resolve this field earlier instead of using `receipt.extraData`, see line 591

Reply:
thanks for the review - fixed.